src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -160,14 +166,33 @@ public void Initialize(string configuration)
             _fullTaskJSON = fullTask.ToString();
 
             _units = units;
-            _oldUnitState = Enumerable.Repeat(ConfigurationUnitState.Unknown, units.Count).ToArray();
+            _oldUnitState = Enumerable.Repeat(string.Empty, units.Count).ToArray();
         }
+    }
+
+    private void HandleEndState(TaskJSONToCSClasses.BaseClass response, bool isSetSuccessful = false)
+    {
+        List<ApplyConfigurationUnitResult> unitResults = new();
+        for (var i = 0; i < _units.Count; i++)
+        {
+            var task = _units[i];
+            if (isSetSuccessful || response.Tasks[i].Status == "Succeeded")

```

Comment:
to confirm, the total amount of units in `_unit` == the total amount of tasks in `response.Tasks`?

Also about the `isSetSuccessful` , is there any chance that its true while the `response.Tasks[i].Status` failed?

Reply:
That is correct. And no, if one task fails, the set will fail.